File: Programmers/Dynamic Programming/거스름 돈.cpp
#include <string>
#include <vector>
#define MOD 1000000007

using namespace std;

int solution(int n, vector<int> money) {
    vector<int> dp(n + 1);

    dp[0] = 1;
    for (int i = 0; i < money.size(); i++) {
        for (int j = money[i]; j <= n; j++) {
            dp[j] += dp[j - money[i]] % MOD;
        }
    }

    return dp[n];
}

File: Programmers/Dynamic Programming/거스름 돈.py
def solution(n, money):
    MOD = 1000000007
    dp = [0] * (n + 1)

    dp[0] = 1
    for i in range(len(money)):
        for j in range(money[i], n + 1):
            dp[j] += dp[j - money[i]] % MOD

    return dp[n]